sys/include/net/gnrc/ndp2.h

```diff
@@ -0,0 +1,355 @@
+ *
+ * @note    Should only be used with router advertisemnents. This is not checked
+ *          however, since nodes should silently ignore it in other NDP messages.
+ *
+ * @param[in] mtu           The recommended MTU for the link.
+ * @param[in] next          More options in the packet. NULL, if there are none.
+ *
+ * @return  The packet snip list of options, on success
+ * @return  NULL, if packet buffer is full
+ */
+gnrc_pktsnip_t *gnrc_ndp2_opt_mtu_build(uint32_t mtu, gnrc_pktsnip_t *next);
+
+/**
+ * @brief   Send pre-compiled neighbor solicitation depending on a given network
+ *          interface.
+ *
+ * @pre `(tgt != NULL) && !ipv6_addr_is_multicast(tgt)`
+ * @pre `(netif != NULL) && (dst != NULL)`
+ *
+ * @param[in] tgt   The target address of the neighbor solicitation.
+ *                  May not be NULL and **MUST NOT** be multicast.
+ * @param[in] netif Interface to send over. May not be NULL.
+ * @param[in] src   Source address for the neighbor solicitation. Will be chosen
+ *                  from the interface according to @p dst, if NULL.
+ * @param[in] dst   Destination address for neighbor solicitation. May not be
+ *                  NULL.
+ */
+void gnrc_ndp2_nbr_sol_send(const ipv6_addr_t *tgt, gnrc_ipv6_netif_t *netif,
+                            const ipv6_addr_t *src, const ipv6_addr_t *dst);
+
+/**
+ * @brief   Send pre-compiled neighbor advertisement depending on a given
+ *          network interface.
+ *
+ * @pre `(tgt != NULL) && !ipv6_addr_is_multicast(tgt)`
+ * @pre `(netif != NULL) && (dst != NULL)`
+ *
+ * If @p netif is a forwarding interface and router advertisements are
+ * activated the @ref NDP_NBR_ADV_FLAGS_R is set in the neighbor advertisement.
+ * If @p dst is the unspecified address it will be replaced with ff02::1 and
+ * @p supply_tl2a is set to true implicitly. Otherwise, the
+ * @ref NDP_NBR_ADV_FLAGS_S will be set. If @p tgt is an anycast address
+ * on @p netif the @ref NDP_NBR_ADV_FLAGS_O flag will be set.
+ *
+ * The source address of the IPv6 packet will be left unspecified, so the
+ * @ref net_gnrc_ipv6 "IPv6 module" selects a fitting IPv6 address.
+ *
+ * @param[in] tgt           Target address for the neighbor advertisement. May
+ *                          not be NULL and **MUST NOT** be multicast.
+ * @param[in] netif         Interface to send over. May not be NULL.
+ * @param[in] dst           Destination address for neighbor advertisement. May
+ *                          not be NULL. Is set to ff02::1 when equal to ::
+ *                          (to allow for simple reply mechanisms to neighbor
+ *                          solicitations). This also implies that
+ *                          @p supply_tl2a **must** be true and the parameter
+ *                          will be reset accordingly. If @p dst is not ::, the
+ *                          @ref NDP_NBR_ADV_FLAGS_S flag will be set.
+ * @param[in] supply_tl2a   Add target link-layer address option to neighbor
```

Comment:
There's a case in the implementation of `gnrc_ndp2_nbr_adv_send`, where this bool gets overridden. Could you drop more information on this case in this doc?